Mathlib/Combinatorics/SimpleGraph/EdgeLabelling.lean

```diff
@@ -0,0 +1,221 @@
+  by rw [get]
+
+variable {C : EdgeLabelling G K}
+
+theorem get_comm (x y : V) (h) : C.get y x h = C.get x y h.symm := by
+  simp only [EdgeLabelling.get, Sym2.eq_swap]
+
+@[ext]
+theorem ext_get {C' : EdgeLabelling G K}
+    (h : ∀ x y, (h : G.Adj x y) → C.get x y h = C'.get x y h) : C = C' := by
+  refine funext fun ⟨e, he⟩ => ?_
+  induction e using Sym2.inductionOn
+  exact h _ _ he
+
+/-- Compose an edge-labelling with a function on the colour set. -/
+def compRight (C : EdgeLabelling G K) (f : K → K') : EdgeLabelling G K' :=
+  f ∘ C
+
+/-- Compose an edge-labelling with a graph embedding. -/
+def pullback (C : EdgeLabelling G K) (f : G' ↪g G) : EdgeLabelling G' K :=
+  C ∘ f.mapEdgeSet
+
+@[simp]
+theorem pullback_apply {f : G' ↪g G} e : C.pullback f e = C (f.mapEdgeSet e) :=
+  rfl
+
+@[simp]
+theorem pullback_get {f : G' ↪g G} (x y) (h : G'.Adj x y) :
+    (C.pullback f).get x y h = C.get (f x) (f y) (by simpa) :=
+  rfl
+
+@[simp]
+theorem compRight_apply (f : K → K') (e) : C.compRight f e = f (C e) :=
+  rfl
+
+@[simp]
+theorem compRight_get (f : K → K') (x y) (h : G.Adj x y) :
+    (C.compRight f).get x y h = f (C.get x y h) :=
+  rfl
+
+attribute [scoped instance] Sym2.Rel.setoid in
+/-- Construct an edge labelling from a symmetric function taking values everywhere except the
+diagonal.
+-/
+def mk (f : ∀ x y : V, G.Adj x y → K)
+    (f_symm : ∀ (x y : V) (H : G.Adj x y), f y x H.symm = f x y H) : EdgeLabelling G K :=
+  fun ⟨e, he⟩ => by
+    revert he
+    refine Quotient.hrecOn e (fun xy => f xy.1 xy.2) ?_
+    rintro ⟨a, b⟩ ⟨c, d⟩ ⟨⟩
+    · rfl
+    refine Function.hfunext ?_ ?_
+    · ext; simp only [adj_comm]
+    · intro h₁ h₂ _
+      exact heq_of_eq (f_symm _ _ _)
+
+theorem mk_get (f : ∀ x y : V, G.Adj x y → K) (f_symm) (x y : V) (h : G.Adj x y) :
+    (EdgeLabelling.mk f f_symm).get x y h = f x y h :=
+  rfl
+
+/--
+Given an edge labelling and a choice of label `k`, construct the graph corresponding to the edges
+labelled `k`.
+-/
+def labelGraph (C : EdgeLabelling G K) (k : K) : SimpleGraph V :=
+  SimpleGraph.fromEdgeSet {e | ∃ h : e ∈ G.edgeSet, C ⟨e, h⟩ = k}
+
+theorem labelGraph_adj {C : EdgeLabelling G K} {k : K} (x y : V) :
+    (C.labelGraph k).Adj x y ↔ ∃ H : G.Adj x y, C ⟨s(x, y), H⟩ = k := by
+  rw [EdgeLabelling.labelGraph]
+  simp only [mem_edgeSet, fromEdgeSet_adj, Set.mem_setOf_eq, Ne.eq_def]
+  apply and_iff_left_of_imp _
+  rintro ⟨h, -⟩
+  exact h.ne
+
+instance [DecidableRel G.Adj] [DecidableEq K] (k : K) {C : EdgeLabelling G K} :
+    DecidableRel (C.labelGraph k).Adj := fun _ _ =>
+  decidable_of_iff' _ (EdgeLabelling.labelGraph_adj _ _)
+
+theorem labelGraph_le (C : EdgeLabelling G K) {k : K} : C.labelGraph k ≤ G := by
+  intro x y
+  rw [EdgeLabelling.labelGraph_adj]
+  rintro ⟨h, -⟩
+  exact h
+
+theorem pairwiseDisjoint_univ_labelGraph {C : EdgeLabelling G K} :
+    Set.PairwiseDisjoint (@Set.univ K) C.labelGraph := by
```

Comment:
What about
```suggestion
theorem pairwise_disjoint_labelGraph {C : EdgeLabelling G K} :
    Pairwise fun k l ↦ Disjoint (C.labelGraph k) (C.labelGraph l) := by
```

Reply:
I'd say let's have both here, sonetimes the PairwiseDisjoint spelling is more useful